test/components/routers/test_transformers_text_router.py

```diff
@@ -0,0 +1,131 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+from unittest.mock import patch, MagicMock
+
+import pytest
+
+from haystack.components.routers.transformers_text_router import TransformersTextRouter
+from haystack.utils import ComponentDevice, Secret
+
+
+class TestTransformersTextRouter:
+    @patch("haystack.components.routers.transformers_text_router.AutoConfig.from_pretrained")
+    def test_to_dict(self, mock_auto_config_from_pretrained):
```

Comment:
Let's add a to_dict and from_dict test with the device passed directly to the TransformersTextRouter as a parameter.

Reply:
I added two new tests!